### ISSUE TITLE
Editing institution's department bugs

### ISSUE THREAD

**Issue by plakitkelly** [keeleinstituut/tv-tolkevarav#492]
- [ ] user with add_department/delete_department cannot add/delete without edit_department privilege - muuda button is not visible in department section 
- [ ] if user adds two departments with same name, user sees "UUENDATUD" in previous row, another field has red border and user gets yellow notification "Üksus ... edukalt uuendatud". Usually success message is green. I think it's related with Maret's comment: it is not easy to show successes and errors at the same time [(here](https://github.com/keeleinstituut/tv-tolkevarav/issues/180#issuecomment-1673112822) ) - One department has been saved. And usually after successfully dding the department the modal closes, but in this case the modal is still open.
![image](https://github.com/keeleinstituut/tv-tolkevarav/assets/130659408/d439fcff-1f72-4651-8d6e-34d1f7de8504)
![image](https://github.com/keeleinstituut/tv-tolkevarav/assets/130659408/34565630-8026-4289-bad1-15c5a4f64945)

- [ ] If user adds new department with name that is previously added, nothing happens. Only red border 
![image](https://github.com/keeleinstituut/tv-tolkevarav/assets/130659408/19a1afb5-c6db-4ca9-86c7-79010077ef5c)
Used don't get any error.

- [ ] If user adds 4 or more departments with same name, user gets two notifications
![image](https://github.com/keeleinstituut/tv-tolkevarav/assets/130659408/6624bf05-a774-405c-a01f-ccf187c54677)
![image](https://github.com/keeleinstituut/tv-tolkevarav/assets/130659408/59d57703-835b-4272-8d61-89e5cf09e437)

- [ ] If user adds 3 departments with same name, user gets only yellow notification "üksus edukalt uuendatud"

To do:
[wiki](https://github.com/keeleinstituut/tv-tolkevarav/wiki/6.-Asutuse-peakasutaja-vaade#61-flow-adding--editing--deleting-institutions-%C3%BCksused-departments)
[Figma](https://www.figma.com/file/VqCwED3Dymn7L1gGoAOK27/EKI-t%C3%B5lkev%C3%A4rav?type=design&node-id=4603-101632&mode=design&t=RMIBlZeUJvwIjK3q-4)
- [ ] Add delete department confirmation modal

**Comment by MariusJulius** [comment]
Blocked by: https://github.com/keeleinstituut/tv-tolkevarav/issues/439

**Comment by plakitkelly** [comment]
Is there any business rule for department name? If no, you should add at least max allowed length. 

- I have text with 9999 chars. 

![image](https://github.com/keeleinstituut/tv-tolkevarav/assets/130659408/dcbc5f00-e604-43b5-9d2b-261e51d17a0a)

- "Muuda" button moves to the end of the text. If I entered very long name without spaces, then I couldn't see "Muuda" button anymore and the page was not scrollable either. I was not able to click the "Muuda" button (I had to use dev tool)

![image](https://github.com/keeleinstituut/tv-tolkevarav/assets/130659408/73b674f0-65b9-40e6-8a8a-53d353a00e0d)

**Comment by MariusJulius** [comment]
Moved one item to a separate issue: https://github.com/keeleinstituut/tv-tolkevarav/issues/777
